service/storage_service.cc

```diff
@@ -3076,17 +3076,17 @@ void storage_service::add_expire_time_if_found(inet_address endpoint, int64_t ex
     }
 }
 
-void storage_service::shutdown_client_servers() {
-    for (auto& [name, hook] : _client_shutdown_hooks) {
-        slogger.info("Shutting down {}", name);
+void storage_service::shutdown_protocol_servers() {
+    for (auto& server : _protocol_servers) {
+        slogger.info("Shutting down {} server", server->name());
         try {
-            hook();
+            server->stop_server().get();
```

Comment:
transport/ is not the same as cql3::query_processor. Note that thrift also uses query_processor. transport/ is only about managing the tcp port and protocol processing, and other protocols do not make calls on the same port.

Reply:
@avikivity it's a good point that the server **object** and its **listening socket** aren't the same thing, but in many of these cases it's not clear exactly what closing the "protocol server" will actually do - it might destroy both. If we decide that a "protocol server" only holds the listening sockets and processes protocol - and does hold other objects and those are kept separately, you statement is correct, but I'm not sure at all that this is what we're doing. You definitely can't make Alternator calls (not just through the socket) after closing the Alternator server, I don't know about CQL.